src/uu/head/src/head.rs

```diff
@@ -293,9 +304,20 @@ fn read_but_last_n_bytes(input: &mut impl std::io::BufRead, n: usize) -> std::io
 
 fn read_but_last_n_lines(
     input: impl std::io::BufRead,
-    n: usize,
+    n: u64,
     separator: u8,
 ) -> std::io::Result<()> {
+    let n = match usize::try_from(n) {
```

Comment:
as it seems to be the same, maybe create a function for this ?

Reply:
ping ?

Reply:
ok, i'll do some more experiments :-)